Curvature/HTTPRequest/HTTPRequest.swift

```diff
@@ -32,6 +32,7 @@ public struct HTTPRequest {
 
     public var parameters: [String: String] = [:]
     public var data: [String: Any] = [:]
+    public var context: [String: Any] = [:]
```

Comment:
Yeah, actually I was thinking about `data` as a `context`, really. You push custom data to the request using keys which should be documented in the middlewares and on the other side you extract the data using that key. Camarilla for example does that, it puts a `JSON` object inside the dictionary using the key "JSON". Then it provides an computed property to abstract the retrieval of the value from the dictionary. If two middlewares use the same key. The last one will overwrite the first. That's something that should be stated very clearly in the docs.

Now, about the new concept of `data`:

> I think the data variable should represent a deserialised dictionary of data parsed from the request body.

This might be a good idea. I'll try this concept with Camarilla. I only worry that the user would have two options to retrieve the parsed data from the request and might get confused.